src/server/worker/common/common.go

```diff
@@ -88,3 +91,18 @@ dataFilters:
 	}
 	return matchesData
 }
+
+// TODO: Trim non-meta file shards?
+func Shard(pachClient *client.APIClient, fileSetIDs []string) ([]*pfs.PathRange, error) {
```

Comment:
Yeah, I will do a pass on the naming. I agree that everything being called `Shard` is probably not the right move. This particular function grabs the shards for the file set with the most shards. This should work pretty well for the datum file sets since it will ensure enough shards are created for the total work being done.